client/Scenarios/CheckResourceGroupOpsAsync.cs

```diff
@@ -0,0 +1,173 @@
+﻿using Azure.ResourceManager.Core;
+using azure_proto_compute;
+using azure_proto_network;
+using System;
+using System.Runtime.InteropServices.WindowsRuntime;
+
+namespace client
+{
+    class CheckResourceGroupOpsAsync : Scenario
+    {
+        public CheckResourceGroupOpsAsync() : base() { }
+
+        public CheckResourceGroupOpsAsync(ScenarioContext context) : base(context) { }
+
+        public override void Execute()
+        {
+            ExecuteAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        private async System.Threading.Tasks.Task ExecuteAsync()
+        {
+            var client = new AzureResourceManagerClient();
+            var subscription = client.GetSubscriptionOperations(Context.SubscriptionId);
+
+            // Create Resource Group
+            Console.WriteLine($"--------Start create group {Context.RgName}--------");
+            var resourceGroup = subscription.GetResourceGroupContainer().Construct(Context.Loc).CreateOrUpdate(Context.RgName).Value;
+            CleanUp.Add(resourceGroup.Id);
+            var rgOps = subscription.GetResourceGroupOperations(Context.RgName);
+
+            try 
+            {
+                rgOps.AddTag("", "");
+            }
```

Comment:
you don't expect to make it to the next line, so you should throw right after line 33.

Also you can extract this into a method, take a lambda, and a message and exception type, will make this test much more readable